src/cdk/scrolling/virtual-scroll-viewport.ts

```diff
@@ -66,10 +63,9 @@ export class CdkVirtualScrollViewport implements OnInit, OnDestroy {
   // depending on how the strategy calculates the scrolled index, it may come at a cost to
   // performance.
   /** Emits when the index of the first element visible in the viewport changes. */
-  @Output() scrolledIndexChange: Observable<number> =
-      Observable.create(observer => this._scrollStrategy.scrolledIndexChange
-          .pipe(sample(this._changeDetectionComplete))
-          .subscribe(observer));
+  @Output() scrolledIndexChange: Observable<number> = Observable.create(observer =>
+      this._scrollStrategy.scrolledIndexChange.subscribe(index =>
+          Promise.resolve().then(() => this._ngZone.run(() => observer.next(index)))));
```

Comment:
AFAIK you shouldn't need the `NgZone.run` here due to the `Promise.resolve`.

Reply:
Is this the case even if `scrolledIndexChange` emits outside the `NgZone`? because I think it does

Reply:
I confirmed that it does need the `.run` in this case